### ISSUE TITLE
Fast Refresh performing a full reload when there is a client component + tailwind

### ISSUE THREAD

**Issue by TheBinaryGuy** [vercel/next.js#47533]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.4.0: Mon Mar  6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103
    Binaries:
      Node: 19.0.0
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 7.30.3
    Relevant packages:
      next: 13.2.5-canary.18
      eslint-config-next: 13.2.5-canary.18
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true)

### Link to the code that reproduces this issue

https://stackblitz.com/edit/vercel-next-js-rz1adm

### To Reproduce

- Open page.tsx
- Make a change
- Save
- Sometimes it only happens after 2 or 3 change and save cycle

### Describe the Bug

When using a client component + tailwind Fast Refresh performs a full reload on changes. The issue does not appear if I remove either the styles or the client component. Also, using CSS without tailwind doesn't force full reload. This issue sometimes only appears after 2 or 3 change - save cycle.

https://user-images.githubusercontent.com/13674136/227739045-3c2a153e-a65d-4c04-8dc2-c0ca1abfb7af.mov

### Expected Behavior

I was expecting to see only the changes being made reflected as opposed to a full reload.

**Comment by unbelivr** [comment]
Same issue on v13.2.5-canary.22
Fast refresh starts working when I remove `globals.css` import from `/app/layout.tsx`. 

I rolled back to 13.2.4, everthing is good

**Comment by TheBinaryGuy** [comment]
Closing, #47913 fixes this. Thanks @shuding!

**Comment by mitchatdoshii** [comment]
Just noting here I'm seeing the same behaviour on canary branch - Fast Refresh works fine until I have any `use client` in the app, then performs a full page reload on any stylesheet change (I'm using Sass). 

Rolling back to 13.2.4 also fixes this for me.

**Comment by TheBinaryGuy** [comment]
@mitchatdoshii Yep, you're right, is there another issue or should I reopen this?

**Comment by mitchatdoshii** [comment]
No other open issue I'm aware of - I'd say perhaps reopen this one 😊

**Comment by TheBinaryGuy** [comment]
@mitchatdoshii Yep I couldn't as well, reopening.

Hey @shuding, just a heads up, #47913 fixed this at one point, but it's broken again.

**Comment by andreferreiradaweb** [comment]
I had the same issue, i just deleted the .next directory, node_modules and update my node version to v18.16.1 and it worked for me.

**Comment by mitchatinvestin** [comment]
Just noting (it's @mitchatdoshii just under a different account), this is still occurring with canary. 

Seems like every second update of Next fixes it, and every subsequent one breaks it again lol

**Comment by atsixian** [comment]
I had the same issue on Next.js 14.2.14. Removing `@tailwind` directives from `global.css` works for me. It's a hack, would be great to find the root cause.
https://stackoverflow.com/questions/67600655/next-js-with-tailwind-refreshes-slow/70742247#70742247

**Comment by marcoshernanz** [comment]
Thanks for the report. I reproduced the full reload on 13.2.5-canary.18 and traced it to #45787. #47913 fixed this regression in 13.2.5-canary.30 and 13.3.0.

I verified the original client component plus Tailwind case through repeated edits on 16.3.2 and 16.4.0-canary.4 with both webpack and Turbopack. Client state is preserved. Related CSS HMR reports now have different reproduction conditions and separate issues, so I’m closing this as fixed.

If this exact case still fails on a current release, please open a new issue with an updated reproduction.
